src/v/archival/archival_policy.cc

```diff
@@ -431,8 +411,7 @@ archival_policy::get_next_compacted_segment(
   ss::shared_ptr<storage::log> log,
   const cloud_storage::partition_manifest& manifest,
   ss::lowres_clock::duration segment_lock_duration) {
-    auto plog = get_concrete_log_impl(std::move(log));
-    if (!plog) {
+    if (log->segment_count() == 0) {
         vlog(
           archival_log.warn,
           "Upload policy find next compacted segment: cannot find log for ntp: "
```

Comment:
Should this warning message change, as the condition has changed?

Reply:
thanks. fixed